File: message/message-java/src/test/java/quan/message/user/UserInfo.java
package quan.message.user;

import java.util.*;
import java.io.IOException;
import quan.message.*;

/**
 * Created by 自动生成
 */
public class UserInfo extends Bean {

    //ID
    private long id;

    //名字
    private String name = "";

    //等级
    private int level;

    public UserInfo() {
    }

    /**
     * ID
     */
    public long getId() {
        return id;
    }

    /**
     * ID
     */
    public UserInfo setId(long id) {
        this.id = id;
        return this;
    }

    /**
     * 名字
     */
    public String getName() {
        return name;
    }

    /**
     * 名字
     */
    public UserInfo setName(String name) {
        Objects.requireNonNull(name);
        this.name = name;
        return this;
    }

    /**
     * 等级
     */
    public int getLevel() {
        return level;
    }

    /**
     * 等级
     */
    public UserInfo setLevel(int level) {
        this.level = level;
        return this;
    }

    @Override
    public void encode(Buffer $buffer) throws IOException {
        super.encode($buffer);

        $buffer.writeLong(id);
        $buffer.writeString(name);
        $buffer.writeInt(level);
    }

    @Override
    public void decode(Buffer $buffer) throws IOException {
        super.decode($buffer);

        id = $buffer.readLong();
        name = $buffer.readString();
        level = $buffer.readInt();
    }

    @Override
    public String toString() {
        return "UserInfo{" +
                "id=" + id +
                ",name='" + name + '\'' +
                ",level=" + level +
                '}';

    }

}

File: message/message_cs/test/user/UserInfo.cs
using System;
using System.Collections.Generic;
using message_cs;
using Buffer = message_cs.Buffer;

namespace message_cs.test.user
{
	/// <summary>
	/// Created by 自动生成
	/// </summary>
    public class UserInfo : Bean
    {
		public long id { get; set; }

		private string _name = "";

		public string name
		{
	    	get => _name;
	    	set => _name = value ?? throw new NullReferenceException();
		}

		public int level { get; set; }


		public UserInfo()
		{
		}

		public override void Encode(Buffer buffer)
		{
	    	base.Encode(buffer);

		    buffer.WriteLong(id);
		    buffer.WriteString(name);
		    buffer.WriteInt(level);
		}

		public override void Decode(Buffer buffer)
		{
	    	base.Decode(buffer);

		    id = buffer.ReadLong();
		    name = buffer.ReadString();
		    level = buffer.ReadInt();
		}

		public override string ToString()
		{
			return "UserInfo{" +
					"id=" + id +
					",name='" + name + '\'' +
					",level=" + level +
					'}';
		}
    }
}